### ISSUE TITLE
add a complexity check

### ISSUE THREAD

**Issue by mgenev** [artificialio/sane#136]
Add this for checking the programmer's code complexity 

https://www.npmjs.com/package/complexity-report

**Comment by IanVS** [comment]
Is that much different from http://eslint.org/docs/rules/complexity?

**Comment by mgenev** [comment]
I think it's the same thing under the hood

**Comment by mgenev** [comment]
Do we have that already?

**Comment by IanVS** [comment]
Yes, we would just need to turn on the rule.

**Comment by mgenev** [comment]
Oh, cool. Is it facing the code of sane, or the code that the dev writes within the stack?

**Comment by IanVS** [comment]
Code of SANE. ESLint runs as part of the mocha suite through mocha-eslint.

**Comment by mgenev** [comment]
I see, what I mean here is to add the tool for the dev to tell them if their app's code is clean and testable or getting out of control. Would be a good feature of the stack. Seems like the escomplex does a pretty robust report. We can play with it and see how it goes. I think on the ember site it may have to become a ember-cli-plugin

**Comment by IanVS** [comment]
I see.  Seems to me like it would be best as an addon, is that what you were thinking?

**Comment by mgenev** [comment]
addon is probably best

**Comment by Globegitter** [comment]
Yep there is https://github.com/ember-cli/ember-cli/issues/3091 that I have looked into recently. I would actually prefer to have eslint over jshint on the ember-cli side, so having that as an addon would be great. And for sails we would just need a `sails-hook-eslint` which shouldn't be too hard to write.

Afterwards it would be equally easy to wrap this into a sane-addon.

Also @IanVS I think we should activate that for sane itself as well. Seems like a good measure.

**Comment by mgenev** [comment]
Looks like ember-cli itself is using a tool called code climate for this if you look at their badges.

**Comment by Globegitter** [comment]
@mgenev Yep I was looking at that as well a couple of weeks ago. Unfortunately they don't support ES2015/ES7 properly yet. So delivers bogus results for at least the sane codebase.

**Comment by IanVS** [comment]
@Globegitter, it looks like linting is now pluggable since https://github.com/ember-cli/ember-cli/pull/3122. I haven't looked into how to swap out JSHint yet, but I agree it would be nice to use ESLint throughout.

**Comment by Globegitter** [comment]
Just found one piece of the puzzle: https://github.com/jonathanKingston/ember-cli-eslint
